### PR TITLE
ASoC: Intel: sof_sdw: increase sdw pin index for each sdw link

### PR DESCRIPTION
**Pull request by bardliao**
The existing code always set sdw cpu dai name as "SDW%d Pin2" for playback and "SDW%d Pin3"for capture. This only work if there is only one dai link for each sdw link. We should increase the sdw pin index for each sdw link.
For example, the cpu dai names should be "SDW0 Pin2", "SDW0 Pin3", and "SDW0 Pin4" for SDW0-Playback-SimpleJack, SDW0-Capture-SimpleJack, and SDW0-Playback-SmartAmp links.

### CODE REVIEW

File: sound/soc/intel/boards/sof_sdw.c [hunk 1]
```diff
@@ -22,7 +22,16 @@ MODULE_PARM_DESC(quirk, "Board-specific quirk override");
...
 #define INC_ID(BE, CPU, LINK)	do { (BE)++; (CPU)++; (LINK)++; } while (0)
 
+/* To store SDW Pin index for each SoundWire link */
+static unsigned int sdw_pin_index[] ={
+	SDW_INTEL_BIDIR_PDI_BASE,
+	SDW_INTEL_BIDIR_PDI_BASE,
+	SDW_INTEL_BIDIR_PDI_BASE,
+	SDW_INTEL_BIDIR_PDI_BASE,
+};
+
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c:29]: The commit message is not clear, suggested edits:

To support multiple codecs per SoundWire link, we have to assign multiple CPU DAIs to different DAI links sharing the same physical link. This is not possible with the existing code since we assume that only 'Pin2' is used for playback and 'Pin3' used for capture - additional DAIs cannot be handled.

This patch enables more CPU DAIs so be used, e.g. "SDW0 Pin2", "SDW0 Pin3",
and "SDW0 Pin4" for SDW0-Playback-SimpleJack, SDW0-Capture-SimpleJack,
and SDW0-Playback-SmartAmp DAI links on physical link #0.

File: sound/soc/intel/boards/sof_sdw.c [hunk 2]
```diff
@@ -22,7 +22,16 @@ MODULE_PARM_DESC(quirk, "Board-specific quirk override");
...
 static void log_quirks(struct device *dev)
+
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: spurious line

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/intel/boards/sof_sdw.c [hunk 3]
```diff
@@ -1483,6 +1487,9 @@ static int sof_card_dai_links_create(struct device *dev,
 	for (i = 0; i < SDW_MAX_GROUPS; i++)
 		group_generated[i] = false;
 
+	for (i = 0; i < SDW_MAX_LINKS; i++)
+		sdw_pin_index[i] = SDW_INTEL_BIDIR_PDI_BASE;
```
Comment (ranj063) [inline sound/soc/intel/boards/sof_sdw.c:1491]: This is so hard to follow. This isnt really pin index is all links have the same value. Its really link_pin_index_base, right? and if so, why do they all have the same value?

Comment (bardliao) [reply sound/soc/intel/boards/sof_sdw.c:1491]: > This is so hard to follow. This isnt really pin index is all links have the same value. Its really link_pin_index_base, right? and if so, why do they all have the same value?

This is to set the initial value for each link. And the pin index base are all SDW_INTEL_BIDIR_PDI_BASE for all links. In other words, the pin index always starts from 2 for all links.
For example,
SDW0 Pin2 for jack playback
SDW0 Pin3 for jack capture
SDW1 Pin2 for speaker playback
SDW1 Pin3 for speaker feedback capture
SDW3 Pin2 for DMIC

**Review by ranj063 — APPROVED** [review]

File: sound/soc/intel/boards/sof_sdw.c (hunk 3 above)
Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c:1491]: to be clearer @ranj063: Pin0/1 are reserved for "bulk", a SoundWire mechanism where command data can be transmitted over data ports. All audio transfers are handled starting with Pin2+.

**Review by plbossart — APPROVED** [review]
